docs/design/sbom.md

```diff
@@ -0,0 +1,84 @@
+Database layout for storing SBOM information.
+
+When storing an SBOM, we extract parts of it into the database, and also store the full document should we need it in
+the future.
+
+We do extract the following components out of the SBOM:
+
+* Basic document metadata
+* Packages
+* Relationships between packages
+
+We also detect PURLs and CPEs, and create database entries for those as well if they don't exist yet.
+
+An SBOM contains a list of packages, which are connected by relationships (like A depends on B). For some SBOM formats
+(e.g. SPDX), this may include references to the document itself (A describes SBOM), or to external documents
+(SBOM-A amends SBOM-B) and packages in external documents (A depends on SBOM-B/C).
+
+To represent this graph of SBOM, packages and relationships, the model of SPDX is being used, as it also covers the
+features of CycloneDX. In the database, this is being represented by the following structure:
+
+```mermaid
+erDiagram
+    Sbom {
+        int sbom_id PK "inherited"
+        string id "inherited"
+        string name "inherited"
+        string namespace "inherited"
+        uuid sha256
```

Comment:
Seems like a lot of id's: sbom_id, id, and sha256. There's also a node_id and document_id in the entity. I'm confused.